### PR TITLE
Refactor to move client list to AllowedClients

### PR DESCRIPTION
**Pull request by RyanMacG**
This provides a clear seperation of concerns between requests and
retrieving the actual data.

It also massively slims down the sintra app/router while maintaining the
existing functionality.

I've also fixed the linting command to use docker and sorted all the existing
failures out.

### CODE REVIEW

File: spec/lib/allowed_clients_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ENV['FRONTEND_API_KEY'] = 'supersecret'
+  end
+
+  after do
+    db[:site].truncate
+    db[:siteip].truncate
+  end
+
+  context 'with an API key' do
+    context 'with the correct API key' do
```
Comment (emileswarts) [inline spec/lib/allowed_clients_spec.rb]: Can probably remove references to the API key from this spec completely

File: spec/clients_spec.rb [hunk 2]
```diff
@@ -29,23 +29,23 @@
...
+        it 'returns an empty string' do
           ENV.delete('FRONTEND_API_KEY')
           get '/clients'
           expect(last_response.body).to eq('')
         end
       end
 
       context 'Given the API key is empty' do
-        it 'raises an error' do
+        it 'returns an empty string' do
```
Comment (emileswarts) [inline spec/clients_spec.rb]: nice 1 for fixing this up

Comment (emileswarts) [reply spec/clients_spec.rb]: Also would say remove knowledge of the allowed clients from this spec (the specific return values of that object), and just assert that it was called or not

Comment (RyanMacG) [reply spec/clients_spec.rb]: @emileswarts done

**Pull request merged**